### PR TITLE
refactor: Migrated CRA to Vite

### PR DESCRIPTION
**Pull request by HanzCEO**
Reason to migrate:
- [x] Faster development than CRA (server startup, hot-reload, builds)
- [x] Production build is only one `.js` file (except static assets)
- [x] Builds with swc (faster build)
- [x] No more dependency vulnerability.
image
- [x] 400% Faster GitHub Actions
image

### CODE REVIEW

**Comment by HanzCEO** [comment]
Hi @scraly,
I have been working on improvements on this repo by depending on this very PR in order to make the website more maintainable and extendable (I'm trying to solve #531). The current design is hard to maintain and I want to extend the capability into something more modular like this:

image
image
image

I have searched for libraries to implement these complex views. But overall, this would be more maintainable and I won't be using many hacks like the current design. Perhaps, I need a separate branch from `main`? Something like `new-design`

**Comment by scraly** [comment]
Hi 🙂
I like the new design proposition! 
I think users of this website have different needs.
I will review your PR with vite integration and then you can work on new design 💪

Yes in a new branch or will be good at a start

File: .github/workflows/ghpages.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           cache: npm
           cache-dependency-path: '**/package-lock.json'
       - working-directory: page
-        run: npm install -D --force
```
Comment (gouz) [inline .github/workflows/ghpages.yml:23]: Maybe we need to kept this tag

File: page/index.html [hunk 2]
```diff
@@ -19,6 +19,7 @@
   </head>
   <body>
     <div id="root"></div>
+    <script type="module" src="/src/main.jsx"></script>
```
Comment (gouz) [inline page/index.html:22]: jsx extension is more for components than main file

**Review by gouz — COMMENTED** [review]
Vite is a better solution ;)

File: .github/workflows/ghpages.yml (hunk 1 above)
Comment (HanzCEO) [reply .github/workflows/ghpages.yml:23]: Would you mind to enlighten me about this?

File: page/index.html (hunk 2 above)
Comment (HanzCEO) [reply page/index.html:22]: I can't seem to export Vite when I change the `main.jsx` file or this script tag to `.js`.
Or am I missing something?

File: .github/workflows/ghpages.yml (hunk 1 above)
Comment (gouz) [reply .github/workflows/ghpages.yml:23]: We got a problem when we update the previous version, the tag --force was mandatory

File: page/index.html (hunk 2 above)
Comment (gouz) [reply page/index.html:22]: yes it seems not working too on my computer ... and I didn't know why

**Review by gouz — APPROVED** [review]

**Comment by scraly** [comment]
awesome, thanks! :)

File: .github/workflows/ghpages.yml (hunk 1 above)
Comment (HanzCEO) [reply .github/workflows/ghpages.yml:23]: I have fixed that problem

**Comment by scraly** [comment]
Thanks again for the PR.
I can't wait to see the new proposals with the new design :)

**Comment by HanzCEO** [comment]
> Thanks again for the PR. I can't wait to see the new proposals with the new design :)
@scraly

I'm currently busy in vacation with my mother,
so I can't guarantee it will be done in July..

But thank you for staying tuned

**Comment by scraly** [comment]
Hi,
any news? 🙏
thanks!

**Comment by HanzCEO** [comment]
I haven't worked on this for a while, I will make sure to put on PR bit-by-bit.

**Pull request merged**